app/models/deal.rb

```diff
@@ -0,0 +1,28 @@
+class Deal < ApplicationRecord
+
+  # Validations
+  validates :title, presence: true, uniqueness: { case_sensitive: false }
```

Comment:
also make sure that we do not fire uniqueness validation query in case empty title is entered by user